programs/mango-v4/src/instructions/openbook_v2_cancel_all_orders.rs

```diff
@@ -0,0 +1,78 @@
+use anchor_lang::prelude::*;
+use openbook_v2::cpi::accounts::CancelOrder;
+
+use crate::accounts_ix::*;
+use crate::error::*;
+use crate::state::*;
+
+pub fn openbook_v2_cancel_all_orders(ctx: Context<OpenbookV2CancelOrder>, limit: u8) -> Result<()> {
+    //
+    // Validation
+    //
+    {
+        let account = ctx.accounts.account.load_full()?;
+        // account constraint #1
+        require!(
+            account
+                .fixed
+                .is_owner_or_delegate(ctx.accounts.authority.key()),
+            MangoError::SomeError
+        );
+
+        let openbook_market = ctx.accounts.openbook_v2_market.load()?;
+
+        // Validate open_orders #2
+        require!(
+            account
+                .openbook_v2_orders(openbook_market.market_index)?
+                .open_orders
+                == ctx.accounts.open_orders.key(),
+            MangoError::SomeError
+        );
+    }
+
+    //
+    // Cancel
+    //
+    let account = ctx.accounts.account.load()?;
+    let account_seeds = mango_account_seeds!(account);
+    cpi_cancel_all_orders(ctx.accounts, &[account_seeds], limit)?;
+
+    // let openbook_market = ctx.accounts.openbook_v2_market.load()?;
+    // let oo_ai = &ctx.accounts.open_orders.as_ref();
+    // let open_orders = load_open_orders_ref(oo_ai)?;
+    // let after_oo = OpenOrdersSlim::from_oo(&open_orders);
+    // emit!(Serum3OpenOrdersBalanceLogV2 {
+    //     mango_group: ctx.accounts.group.key(),
+    //     mango_account: ctx.accounts.account.key(),
+    //     market_index: serum_market.market_index,
+    //     base_token_index: serum_market.base_token_index,
+    //     quote_token_index: serum_market.quote_token_index,
+    //     base_total: after_oo.native_base_total(),
+    //     base_free: after_oo.native_base_free(),
+    //     quote_total: after_oo.native_quote_total(),
+    //     quote_free: after_oo.native_quote_free(),
+    //     referrer_rebates_accrued: after_oo.native_rebates(),
+    // });
+
+    Ok(())
+}
+
+fn cpi_cancel_all_orders(ctx: &OpenbookV2CancelOrder, seeds: &[&[&[u8]]], limit: u8) -> Result<()> {
+    let cpi_accounts = CancelOrder {
+        signer: ctx.account.to_account_info(),
```

Comment:
using the mango account as the signer instead of the group has some advantages (safety!, easier filtering for user-owned oos) and disadvantages (account must be passed, not easy to filter for all mango-related oo accounts).

Reply:
I son't think we can use the group as signer, since the OpenOrdersAccount owner is used in the seeds for the OpenOrdersIndexer. we can't share one indexer for the entire group

Reply:
Ah, I see!